server-reboot/duffy.py

```diff
@@ -0,0 +1,118 @@
+        # all is fine, break out of the loop
+        break
+    except ValueError, ve:
+        print("Failed to parse Duffy response: %s" % (dat))
+    except Error, e:
+        print("An unexpected error occured: %s" % (e))
+
+    retries+=1
+    print("Waiting %d seconds before retrying #%d..." % (retry_delay, retries))
+    time.sleep(retry_delay)
+
+
+# NFS-Ganesha Server (parameters need double escape, passed on ssh commandline)
+server_env="export GERRIT_HOST='%s'" % os.getenv("GERRIT_HOST")
+server_env+=" GERRIT_PROJECT='%s'" % os.getenv("GERRIT_PROJECT")
+server_env+=" GERRIT_REFSPEC='%s'" % os.getenv("GERRIT_REFSPEC")
+server_env+=" YUM_REPO='%s'" % os.getenv("YUM_REPO", "")
+server_env+=" GLUSTER_VOLUME='%s'" % os.getenv("EXPORT")
+server_env+=" ENABLE_ACL='%s'" % os.getenv("ENABLE_ACL", "")
+
+# add the export with environment to ~/.bashrc
+cmd="""ssh -t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@%s '
+tee -a ~/.bashrc' <<< "%s"
+""" % (b['hosts'][0], server_env)
+subprocess.call(cmd, shell=True)
+
+cmd="""ssh -t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@%s '
+	yum -y install curl &&
+	curl %s | bash -
+'""" % (b['hosts'][0], server_script)
+rtn_code=subprocess.call(cmd, shell=True)
+
+
+# check rtn_code and skip client part after failure
+if rtn_code == 0:
+    # NFS-Client (parameters need double escape, passed on ssh commandline)
+    client_env="export SERVER='%s'" % b['hosts'][0]
+    client_env+=" EXPORT='/%s'" % os.getenv("EXPORT")
+    client_env+=" TEST_PARAMETERS='%s'" % os.getenv("TEST_PARAMETERS", "")
+
+    # add the export with environment to ~/.bashrc
+    cmd="""ssh -t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@%s '
+        tee -a ~/.bashrc' <<< "%s"
+        """ % (b['hosts'][1], client_env)
+    subprocess.call(cmd, shell=True)
+
+if rtn_code == 0:
+    versions = [3 , 4.0, 4.1]
+    for version in versions:
+    	client_script = client_script.strip(" ")
+        if client_script.endswith(".py"):
+            interpreter_to_run = "python"
+        elif client_script.endswith(".sh"):
+            interpreter_to_run = "bash"
+        cmd="""ssh -t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@%s '
+	         curl -o client_script %s && %s client_script %.1f
+        '""" % (b['hosts'][1], client_script, interpreter_to_run, version)
+        rtn_code=subprocess.Popen(cmd, shell=True)
+
+        # client setting up time
+	time.sleep(30)
+
+	# rebooting server
+	cmd="""ssh -t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@%s '
+    	    systemctl reboot | bash -
+	'""" % (b['hosts'][0])
+	subprocess.call(cmd, shell=True)
+	
+	# time for rebooting server
+	time.sleep(90)	
+
+	# disable the firewall on server, otherwise the client can not connect
+	cmd="""ssh -t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@%s '
+    	    curl %s | bash -
+	'""" % (b['hosts'][0], server_setup_script)
```

Comment:
Post reboot, cant we just restart nfs-ganesha process which should take care of exporting the gluster volume too?